File: apps/docs/docs/.vitepress/examples/scatter-plot.ts
const catchLocationsArray = [
  { lake: "Lake A", x: 5.26, y: 22.64 },
  { lake: "Lake B", x: 30.87, y: 120.75 },
  { lake: "Lake C", x: 50.01, y: 60.94 },
  { lake: "Lake D", x: 115.13, y: 94.16 },
  { lake: "Lake E", x: 133.05, y: 50.44 },
  { lake: "Lake F", x: 85.99, y: 172.78 },
];

chart(catchLocationsArray)
  .flow(scatter("lake", { x: "x", y: "y" }))
  .mark(circle({ r: 5 }))
  .render(root, {
    w: 500,
    h: 300,
    axes: true,
  });

File: packages/gofish-python/stories/vega_lite/scatter_plot.py
"""Vega-Lite/Simple Scatter Plot — mirrors ScatterPlot.stories.tsx

https://vega.github.io/vega-lite/examples/point_2d.html

Cars dataset: Horsepower vs. Miles_per_Gallon.
"""

from gofish import chart, scatter, circle, log

TITLE = "Vega-Lite/Simple Scatter Plot"


def load_data():
    from stories.data.vega import load_cars
    rows = load_cars()
    return [r for r in rows if r.get("Horsepower") and r.get("Miles_per_Gallon")]


def default(data=None, w=300, h=300):
    if data is None:
        data = load_data()
    return (
        chart(data)
        .flow(
            log("cars before scatter"),
            scatter(by="Name", x="Horsepower", y="Miles_per_Gallon"),
        )
        .mark(circle(r=4, fill="rgba(31, 119, 180, 0.4)", stroke="#1f77b4", strokeWidth=1))
    )
